app/models/user.rb

```diff
@@ -21,16 +21,22 @@ def standard?
     role == 'standard'
   end
 
+  def is_owner_of(wiki)
+    admin? || (premium? && (wiki.user == self || wiki.new_record?))
+  end
```

Comment:
Good catch, I'll remove that - the premium status doesn't relate to ownership of a wiki.